src/Infrastructure/Builder/Yaml/CommandBuilder.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Builder\Yaml;
+
+use SprykerSdk\Sdk\Core\Application\Service\TaskPool;
+use SprykerSdk\Sdk\Core\Domain\Entity\Command;
+use SprykerSdk\Sdk\Core\Domain\Enum\TaskType;
+use SprykerSdk\SdkContracts\Entity\ContextInterface;
+use SprykerSdk\SdkContracts\Entity\ConverterInterface;
+use SprykerSdk\SdkContracts\Entity\TaskInterface;
+
+class CommandBuilder implements CommandBuilderInterface
+{
+    /**
+     * @var \SprykerSdk\Sdk\Core\Application\Service\TaskPool
+     */
+    protected TaskPool $taskPool;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Builder\Yaml\ConverterBuilderInterface
+     */
+    protected ConverterBuilderInterface $converterBuilder;
+
+    /**
+     * @param \SprykerSdk\Sdk\Core\Application\Service\TaskPool $taskPool
+     * @param \SprykerSdk\Sdk\Infrastructure\Builder\Yaml\ConverterBuilderInterface $converterBuilder
+     */
+    public function __construct(TaskPool $taskPool, ConverterBuilderInterface $converterBuilder)
+    {
+        $this->taskPool = $taskPool;
+        $this->converterBuilder = $converterBuilder;
+    }
+
+    /**
+     * @param array $data
+     * @param array $taskListData
+     * @param array<string> $tags
+     *
+     * @return array<int, \SprykerSdk\SdkContracts\Entity\CommandInterface>
+     */
+    public function buildCommands(array $data, array $taskListData, array $tags = []): array
+    {
+        $commands = [];
+
+        if (in_array($data['type'], ['local_cli', 'local_cli_interactive'], true)) {
+            $commands[] = $this->buildCommand(
+                $data,
+                $data['tags'] ?? [],
+                false,
+                $this->converterBuilder->buildConverter($data),
+            );
+        }
+
+        $taskSetCommands = $this->buildTaskSetCommands($data, $taskListData, $tags);
+
+        return array_merge($commands, $taskSetCommands);
+    }
+
+    /**
+     * @param array $data
+     * @param array $taskListData
+     * @param array $tags
+     *
+     * @return array
+     */
+    protected function buildTaskSetCommands(array $data, array $taskListData, array $tags = []): array
+    {
+        $commands = [];
+
+        if ($data['type'] !== TaskType::TASK_SET_TYPE) {
+            return $commands;
+        }
+
+        $existingTasks = $this->taskPool->getTasks();
+        foreach ($data['tasks'] as $task) {
+            $tasksTags = $task['tags'] ?? [];
+            if ($tags && !array_intersect($tags, $tasksTags)) {
```

Comment:
you can skip first part, `!array_intersect()` covers this condition

Reply:
Fixed